Measurements/Measurements_Workbook.ipynb

```diff
@@ -0,0 +1,453 @@
+    "\n",
+    "Follows two possibilities:\n",
+    "1. The qubit is measured as $|1\\rangle$, the qubit is in state $|B\\rangle$, we return `false`.\n",
+    "2. The qubit is measured as $|0\\rangle$, the qubit is in state $|A\\rangle$, we return `true`."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Task 1.5. |00⟩ or |11⟩ ?\n",
+    "\n",
+    "**Input:** Two qubits (stored in an array of length 2) which are guaranteed to be in either the $|00\\rangle$ or the $|11\\rangle$ state.\n",
+    "\n",
+    "**Output:** 0 if the qubits were in the $|00\\rangle$ state, or 1 if they were in the $|11\\rangle$ state. The state of the qubits at the end of the operation does not matter."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "**Solution**"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "operation ZeroZeroOrOneOne (qs : Qubit[]) : Int {\n",
+    "\tif(M(qs[0]== One){\n",
+    "\t\treturn One;\n",
+    "\t}\n",
+    "\telse{\n",
+    "\t\treturn Zero;\n",
+    "\t} \n",
+    "}"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Both of the qubits stored in the qs array are in the same state ( for $|00\\rangle$ each individual qubit is in state $|0\\rangle$, for $|11\\rangle$ each individual qubit is in state $|1\\rangle$).Therefore, if we measure one qubit we will know the state of the other.\n",
+    "In other words, if M(qs[0]) measures the first qubit as 1 we know that the qubits in qs are in state $|11\\rangle$ and M(qs[0]) measures the first qubit as 0 we know that the qubits in qs are in state $|00\\rangle$."
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Task 1.6. Distinguish four basis states.\n",
+    "\n",
+    "**Input:** Two qubits (stored in an array of length 2) which are guaranteed to be in one of the four basis states ($|00\\rangle$, $|01\\rangle$, $|10\\rangle$, or $|11\\rangle$).\n",
+    "**Output:**\n",
+    "•\t0 if the qubits were in the $|00\\rangle$ state,\n",
+    "•\t1 if they were in the $|01\\rangle$ state,\n",
+    "•\t2 if they were in the $|10\\rangle$ state,\n",
+    "•\t3 if they were in the $|11\\rangle$ state.\n",
+    "In this task and the subsequent ones the order of qubit states in task description matches the order of qubits in the array (i.e., $|10\\rangle$ state corresponds to qs[0] in state $|1\\rangle$ and qs[1] in state $|0\\rangle$).\n",
+    "The state of the qubits at the end of the operation does not matter."
+   ]
+  },
+  {
+   "attachments": {},
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "**Solution**"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    " operation BasisStateMeasurement_Reference (qs : Qubit[]) : Int {\n",
```

Comment:
Once you add `%kata` magics, you'll need to name the operations exactly the way they are named in the tasks, otherwise they won't get picked up for testing. This one will have to be `BasisStateMeasurement`